src/schemas.jl

```diff
@@ -863,3 +851,27 @@ macro version(record_type, declared_fields_block)
         nothing
     end
 end
+
+#####
+##### Base overload definitions
+#####
+
+function Base.:(==)(x::T, y::T) where {T<:AbstractRecord}
```

Comment:
Actually this isn't quite right: for parametric row types, this will method won't be hit if the type parameters differ. Using this setup (taken from the tests):
```julia
@schema "test.param" Param
@version ParamV1 begin
    i::(<:Integer)
end
```
we get
```julia
julia> ParamV1(; i=one(Int32)) == ParamV1(; i=one(Int64))
false
```
on this branch but `true` on main.